### PR TITLE
Added Quiet value

### LINKED ISSUE #32097: Provide a quiet mode for local-exec
### Terraform Version

```shell
Terraform v1.3.3                                                                                                                
on linux_amd64                                                                                                                  
+ provider registry.terraform.io/hashicorp/null v3.2.0
```

### Use Cases

I find the echoing of the command that local-exec will execute to be entirely too verbose.  The output from even relatively simple examples can be hard to parse.  Medium- to high-complexity examples are overwhelming.

```
locals {
  test = {
    name = "Spacely Space Sprockets"
    competitor = "Cogswell Cogs"
  }
}

resource "null_resource" "test" {
  provisioner "local-exec" {
    command = <<CMD
      echo ${jsonencode(local.test)}
    CMD
  }
}
```
Produces this output --
```
Plan: 1 to add, 0 to change, 0 to destroy.
null_resource.test: Creating...
null_resource.test: Provisioning with 'local-exec'...
null_resource.test (local-exec):Executing: ["/bin/sh" "-c" "      echo {\"competitor\":\"Cogswell Cogs\",\"name\":\"Spacely Space Sprockets\"}\n"]
null_resource.test (local-exec):{competitor:Cogswell Cogs,name:Spacely Space Sprockets}
null_resource.test: Creation complete after 0s [id=5577006791947779410]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```
The "Executing:" output is completely unnecessary.

### Attempted Solutions

None.  I don't see a way to suppress this.

### Proposal

Provide a `quiet` parameter to local-exec.  By default it is `false` indicating that local-exec should output the command about to be executed (i.e. the behavior remains as it is today).  If it is true, the `Executing: ...` output will be supressed.

```
resource "null_resource" "test" {
  provisioner "local-exec" {
    quiet = true
    command = <<CMD
      echo ${jsonencode(local.test)}
    CMD
  }
}
```
Produces this output --
```
Plan: 1 to add, 0 to change, 0 to destroy.
null_resource.test: Creating...
null_resource.test: Provisioning with 'local-exec'...
null_resource.test (local-exec):{competitor:Cogswell Cogs,name:Spacely Space Sprockets}
null_resource.test: Creation complete after 0s [id=5577006791947779410]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

### References

_No response_

### PR DESCRIPTION
**Pull request by itspooya**
Added Quiet value in order to quiet the output from local-exec

Fixes #32097 

## Target Release

1.4.x

### NEW FEATURES | UPGRADE NOTES | ENHANCEMENTS | BUG FIXES | EXPERIMENTS

 

-

### CODE REVIEW

**Comment by crw** [comment]
Thanks for this submission. Although I cannot commit to having this PR reviewed at this time, we acknowledge your contribution and appreciate it. Thanks again for the submission!

**Comment by itspooya** [comment]
Thank you @crw

**Comment by itspooya** [comment]
@crw 
Hi
Is there anything I can do about this PR?
I wanted to send some other PRs too
Thank you

**Comment by crw** [comment]
@itspooya Thanks, we are running it by the team. If it looks OK to accept, then there may be changes requested during review. 

With regards to other PRs, just be mindful of our contributing guidelines, particularly: https://github.com/hashicorp/terraform/blob/main/.github/CONTRIBUTING.md#proposing-a-change. Your PRs are more likely to be accepted if the solution has already been discussed in the Github issue you are attempting to close. Thanks!

**Comment by itspooya** [comment]
@crw Thanks, I am looking forward to improve and being mindful of your contributing guidelines
Also I had a question regarding Vercel and how to authorize to fix the tests
Best Regards
Pooya

**Comment by itspooya** [comment]
@crw 
Hey I just wanted to check whether this pull request would be merged or not?

**Comment by crw** [comment]
@itspooya This one seems to have got lost in the shuffle. Let me follow up with the team and get back to you.

**Review by kmoe — APPROVED** [review]
Thanks for the PR. One minor change.

File: internal/builtin/provisioners/local-exec/resource_provisioner.go [hunk 1]
```diff
@@ -163,7 +167,14 @@ func (p *provisioner) ProvisionResource(req provisioners.ProvisionResourceReques
 	go copyUIOutput(req.UIOutput, tee, copyDoneCh)
 
 	// Output what we're about to run
-	req.UIOutput.Output(fmt.Sprintf("Executing: %q", cmdargs))
+	// if quiet is not set, output the command we're about to run
```
[the remark this answers is not in the document]
Comment (itspooya) [reply internal/builtin/provisioners/local-exec/resource_provisioner.go]: Removed those lines

**Comment by itspooya** [comment]
@kmoe 
Please check it, Thanks

**Comment by itspooya** [comment]
@crw 
Hello, could you check it please?
Thanks

**Comment by crw** [comment]
It is being discussed by the team. I will see if there is a resolution.

**Review by kmoe — APPROVED** [review]
